components/RadioButton/RadioButton.tsx

```diff
@@ -0,0 +1,44 @@
+import React, {useState} from 'react';
+import {View, Text, Pressable} from 'react-native';
+import {Colors} from '../../utils/colors';
+
+import style from './style';
+import globalStyle from '../../assets/styles/globalStyle';
+
+const RadioButton: React.FC<{
+  data: {value: string}[];
+  onSelect: (option: string) => void;
+  selectedValue?: string;
+}> = ({data, onSelect, selectedValue}) => {
+  const [userOption, setUserOption] = useState<string | null>(
+    selectedValue ?? null,
+  );
+
+  return (
+    <View>
+      {data.map((item, index) => (
+        <Pressable
+          key={index}
+          style={item.value === userOption ? style.selected : style.unselected}
+          onPress={() => {
+            setUserOption(item.value);
+            onSelect(item.value);
+          }}>
```

Comment:
Rule of thumb, if the function passed is not inline and it's on multiple rows, then you should create a constant for it.

```tsx
onPress={() => {
  setUserOption(item.value);
  onSelect(item.value);
}}
```

```tsx
onPress={() => selectOption(item.value)}

const selectOption = (selectedOption: string) => {
  setUserOption(selectedOption);
  onSelect(selectedOption);
}
```